aspnetcore/host-and-deploy/web-farm.md

```diff
@@ -0,0 +1,66 @@
+author: guardrex
+description: Learn how to host multiple instances of an ASP.NET Core app with shared resources in a web farm environment.
+ms.author: riande
+ms.custom: mvc
+ms.date: 07/13/2018
+uid: host-and-deploy/web-farm
+---
+# Host ASP.NET Core in a web farm
+
+By [Luke Latham](https://github.com/guardrex)
+
+This topic describes configuration and dependencies for ASP.NET core apps hosted in a web farm that rely upon shared resources.
+
+## General configuration
+
+<xref:host-and-deploy/index>  
+Learn how to set up hosting environments and deploy ASP.NET Core apps. Configure a process manager on each node of the web farm to automate app starts and restarts. Each node requires the ASP.NET Core runtime. For more information, see the topics in the [Host and deploy](xref:host-and-deploy/index) area of the documentation.
+
+<xref:host-and-deploy/proxy-load-balancer>  
+Learn about configuration for apps hosted behind proxy servers and load balancers, which often obscure important request information.
+
+<xref:host-and-deploy/azure-apps/index>  
+[Azure App Service](https://azure.microsoft.com/services/app-service/) is a [Microsoft cloud computing platform service](https://azure.microsoft.com/) for hosting web apps, including ASP.NET Core. App Service is a fully managed platform that provides automatic scaling, load balancing, patching, and continuous deployment.
+
+## Required configuration
+
+Data Protection and Caching require configuration for apps deployed to a web farm.
+
+### Data Protection
+
+The [ASP.NET Core Data Protection system](xref:security/data-protection/introduction) is used by apps to protect data. Data Protection relies upon a set of cryptographic keys stored in a *key ring*. When the Data Protection system is initialized, it applies [default settings](xref:security/data-protection/configuration/default-settings) that store the key ring locally. Under the default configuration, a unique key ring is stored on each node of the web farm. Consequently, each web farm node can't decrypt data that's encrypted by an app on any other node. The default configuration isn't generally appropriate for hosting apps in a web farm. An alternative to implementing a shared key ring is to always route user requests to the same node. For more information on Data Protection system configuration for web farm deployments, see <xref:security/data-protection/configuration/overview>.
+
+### Caching
+
+Caching is required by [ASP.NET Core Sessions](xref:fundamentals/app-state#session-state) and when [caching responses](xref:performance/caching/response). In a web farm environment, the caching mechanism must share cached items across the web farm's nodes. Caching must either rely upon a common Redis cache, a shared SQL Server database, or a custom caching implementation that shares cached items across the web farm. For more information, see <xref:performance/caching/distributed> and <xref:performance/caching/response>.
```

Comment:
> You said what depends on caching but not what depends on data protection. 

... because everything on the list depends on it. I explicitly listed the two for caching because only two dependent components require it.

Reply:
>  response caching does not support IDistributedCache / multiple nodes,

Looks like the wording of the first sentence in the [Response caching topic](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-2.1#distributed-cache) is poor ...

> ### Distributed Cache

Use a distributed cache to store data in memory when the app is hosted in a cloud or server farm. The cache is shared across the servers that process requests. A client can submit a request that's handled by any server in the group if cached data for the client is available. ASP.NET Core offers SQL Server and Redis distributed caches.

Reply:
Hmm, the whole cache topic hierarchy seems backwards. "Cache responses" should not be the top level node, it should just be Caching. (Other caching technology in ASP.NET Core
)[https://docs.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-2.1#other-caching-technology-in-aspnet-core] should be the TOC for the whole Caching category with response caching being one of the sub nodes.

And yes that Distributed Cache description is subtly wrong.
"Use a distributed cache to store data ~~in memory~~ when the app is hosted in a cloud or server farm. The cache is shared ~~across~~ _by_ the servers that process requests. A client can submit a request ~~that's handled by~~ _to_ any server in the group ~~if cached data for the client is available~~ _and get access to the cached data_. ASP.NET Core offers SQL Server and Redis distributed caches."